server/api/helpers/cache/cache-latest.js

```diff
@@ -0,0 +1,68 @@
+const FAST_TRIES = 18;
+const TRY_TIMEOUT = 10 * 60 * 1000; // 10 mins
+const MAX_TRIES = FAST_TRIES * 3;
+const SLOW_TIMEOUT = TRY_TIMEOUT * 3;
+
+const retryUntilSuccess = async (cacheName, toTry, checkFunc, tries = 1) => {
```

Comment:
i think i would leave it as checkFunc because in the future, we might not just be checking whether it has result or not. might be checking if result is valid or is result is update to date etc.